crates/miden-objects/src/errors.rs

```diff
@@ -324,8 +324,23 @@ pub enum AssetError {
       expected_ty = AccountType::NonFungibleFaucet
     )]
     NonFungibleFaucetIdTypeMismatch(AccountIdPrefix),
-    #[error("{0}")]
-    TokenSymbolError(String),
+    #[error("provided token symbol invalid")]
+    TokenSymbolError(#[source] TokenSymbolError),
+}
+
+// TOKEN SYMBOL ERROR
+// ================================================================================================
+
+#[derive(Debug, Error)]
+pub enum TokenSymbolError {
+    #[error("token symbol value {0} cannot exceed {1}")]
+    ValueTooLarge(u64, u64),
```

Comment:
```suggestion
    #[error("token symbol value {0} cannot exceed {max}", max = TokenSymbol::MAX_ENCODED_VALUE)]
    ValueTooLarge(u64),
```

Nit: We can inline the constant.